internal/confmapprovider/discovery/bundle/bundle.d/receivers/mongodb.discovery.yaml

```diff
@@ -0,0 +1,51 @@
+#####################################################################################
+#                               Do not edit manually!                               #
+# All changes must be made to associated .tmpl file before running 'make bundle.d'. #
+#####################################################################################
+mongodb:
+  enabled: true
+  rule:
+    docker_observer: type == "container" and any([name, image, command], {# matches "(?i)mongo"}) and not (command matches "splunk.discovery")
+    host_observer: type == "hostport" and command matches "(?i)mongo" and not (command matches "splunk.discovery")
+    k8s_observer: type == "port" and pod.name matches "(?i)mongo"
+  config:
+    default:
+      username: splunk.discovery.default
+      password: splunk.discovery.default
+      tls:
+        insecure: true
+        insecure_skip_verify: true
+  status:
+    metrics:
+      - status: successful
+        strict: mongodb.database.count
+        log_record:
+          body: mongodb receiver is working!
+    statements:
+      - status: failed
+        regexp: 'connect: network is unreachable'
+        log_record:
+          append_pattern: true
+          body: The container cannot be reached by the Collector. Make sure they're in the same network.
+      - status: failed
+        regexp: 'connect: connection refused'
+        log_record:
+          append_pattern: true
+          body: The container is refusing mongodb connections.
+      - status: partial
+        regexp: '.* unable to authenticate using mechanism .*'
+        log_record:
+          append_pattern: true
+          body: >-
+              Please ensure your user credentials are correctly specified with
+              `--set splunk.discovery.receivers.mongodb.config.username="<username>"` and
+              `--set splunk.discovery.receivers.mongodb.config.password="<password>"` or
+              `SPLUNK_DISCOVERY_RECEIVERS_mongodb_CONFIG_username="<username>"` and
+              `SPLUNK_DISCOVERY_RECEIVERS_mongodb_CONFIG_password="<password>"` environment variables.
+      - status: partial
+        regexp: '.* failed to fetch index stats metrics: (Unauthorized) not authorized on admin to execute command .*'
+        log_record:
+          append_pattern: true
+          body: >-
+            Make sure the user has been given access admin db with full root access.
+            `db.grantRolesToUser('admin', [{ role: 'readWrite', db: 'admin' }])`
```

Comment:
_full root access_ sounds too much. `readWrite` doesn't sound like reading https://www.mongodb.com/docs/manual/reference/built-in-roles/#mongodb-authrole-readWrite. 

Also, why is `write` needed to fetch the index? Sounds like reading should be enough?

Reply:
I think you're correct on readWrite. I have just given that permission and it works. Let me see if read is enough to do the job or not?

Reply:
I have just verified, it works with `role: read`. I will shortly update the wording.


Successfully added user: {
        "user" : "testUser",
        "roles" : [
                {
                        "role" : "read",
                        "db" : "admin"
                }
        ]
}

Reply:
I should read more carefully. Upstream has already mention the minimum requirements. It works well.
All we need is clustermonitor role.

https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/mongodbreceiver#prerequisites

Reply:
Also, we have to tell BITS and prospective customers the upstream version supported are 4.0+ and 5.0. Both of which I have tested.

Reply:
tested successfully with this user below.

db.createUser(
    {
      user: "testUser1",
      pwd: "testPass1",
      roles: [
         { role: "clusterMonitor", db: "admin" },
      ]
    }
)